webui/src/js/viewModels/ingress-design-view-impl.js

```diff
@@ -209,31 +209,58 @@ function(i18n, accUtils, ko, ArrayDataProvider, BufferingDataProvider, project,
       window.api.ipc.invoke('show-error-message', title, message);
     };
 
-    this.handleEditRoute = (event, context) => {
+    async function getTargetServiceDetails (project) {
+
+      const kubectlExe = k8sHelper.getKubectlExe();
```

Comment:
Personally, I would move most of this logic into a k8sHelper method and then just call that method from this one...